provdebug/DebugTrace.py

```diff
@@ -0,0 +1,33 @@
+from .DebugRecord import DebugRecord
+from typing import Optional, List, Dict
+
+class DebugTrace:
```

Comment:
I introduced this data class because we now need to carry around two disjoint pieces of information per trace
* metadata
* debug records